azkaban-common/src/main/java/azkaban/executor/IFlowRunnerManager.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.executor;
+
+import azkaban.utils.FileIOUtils.LogData;
+
+/**
+ * Defines operations common to all flow execution controllers.
+ */
+public interface IFlowRunnerManager {
```

Comment:
This interface is not needed as FlowContainer does not implement half of the APIs and does not require execId at all.

Reply:
Could you please elaborate on why the interface isn't needed? 
-As I see it FlowContainer and FlowRunnerManager have the same main responsibility: to control a flow execution.
-Trigger action objects from `KillExecutionAction` and `KillJobAction` depend on such flow execution controllers for their functionality. Without the interface then we need to go back to if-else of current DispatchMethod and using  `SERVICE_PROVIDER.getInstance` of Guice which I believe is an anti-pattern and not so much extensible/maintainable code.